rclpy/rclpy/logging_service.py

```diff
@@ -39,27 +47,29 @@ def __init__(self, node):
             self._set_logger_levels, qos_profile=qos_profile_services_default
         )
 
-    def _get_logger_levels(self, request, response):
+    def _get_logger_levels(self, request: GetLoggerLevels.Request,
+                           response: GetLoggerLevels.Response) -> GetLoggerLevels.Response:
         for name in request.names:
             logger_level = LoggerLevel()
             logger_level.name = name
             try:
-                ret_level = rclpy.logging.get_logger_level(name)
+                ret_level: Union[int, LoggingSeverity] = rclpy.logging.get_logger_level(name)
```

Comment:
Does this really need to be a `Union`?  As far as I can tell, [get_logger_level](https://github.com/ros2/rclpy/blob/5cbb110b8ac301d7cdf27768010367dc27f92f59/rclpy/rclpy/logging.py#L58-L60) always returns a `LoggingSeverity`.

Reply:
While get_logger_level always returns a LoggingSeverity ret_level can also be set to int. Mypy prefers an explicit expansion of available types. Without the explicit expansion it correctly identifies ret_level as LoggingSeverity but errors when that type is changed to int. Slightly more info on the type inference can be found [here](https://mypy.readthedocs.io/en/stable/type_inference_and_annotations.html#explicit-types-for-variables).

The error in question
Incompatible types in assignment (expression has type "int", variable has type "LoggingSeverity")

Reply:
> While get_logger_level always returns a LoggingSeverity ret_level can also be set to int.

Oh, that's a bug.  Down below when we get a `RuntimeError`, ret_level should be set to `LoggingSeverity.UNSET`, not 0, which I think should resolve the issue.  There *might* be a further issue when we do the assignment `logger_level.level = ret_level`, but if so we should type annotate that differently, in my opinion.